Challenge Problems - GSSoC/islands_in_a_graph.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+Given:
+A grid of size n x m. Each cell is either Red or Blue. Red is denoted as * and . as Blue.
+The entire grid is surrounded in blue.
+A collection of blue cells surrounded by red on all sides is considered a pool.
+
+Tasks:
+
+Count the total number of pools in the grid.
+Return the size of each of the pools in the grid.
+Time and Space complexities for each of these operations.
+Sample:
+5 4
+
+****
+*..*
+****
+**.*
+..**
+
+Number of pools: 2; their size = 2, 1 
+*/
+#include <iostream>
+#include <vector>
+using namespace std;
+
+// stores the input grid
+char grid[105][105];
```

Comment:
Your program will not work for grid greater than 105×105.
Change it to work for every input.